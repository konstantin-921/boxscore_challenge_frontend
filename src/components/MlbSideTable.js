import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import Table from '@material-ui/core/Table'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import TableCell from '@material-ui/core/TableCell'
import TableBody from '@material-ui/core/TableBody'

const styles = theme => ({
  tableCell: {
    paddingRight: 0,
    paddingLeft: 5,
    textAlign: 'center',
    border: '0.5px solid #e0e0e0',
    borderLeft: 'none',
    backgroundColor: '#f0f0f0',
    '&:last-child': {
      paddingRight: 0
    },
  },
});

class MlbSideTable extends React.Component {
  render() {
    const { classes } = this.props
    const data = this.props.data
    const runsAway = data.away_batter_totals.runs
    const runsHome = data.home_batter_totals.runs
    const hitsAway = data.away_batter_totals.hits
    const hitsHome = data.home_batter_totals.hits
    const errorAway = data.away_errors
    const errorHome = data.home_errors
    return (
      <Grid item xs={3}>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell className={classes.tableCell}>R</TableCell>
              <TableCell className={classes.tableCell}>H</TableCell>
              <TableCell className={classes.tableCell}>E</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            <TableRow>
              <TableCell className={classes.tableCell}>{runsAway}</TableCell>
              <TableCell className={classes.tableCell}>{hitsAway}</TableCell>
              <TableCell className={classes.tableCell}>{errorAway}</TableCell>
            </TableRow>
            <TableRow>
              <TableCell className={classes.tableCell}>{runsHome}</TableCell>
              <TableCell className={classes.tableCell}>{hitsHome}</TableCell>
              <TableCell className={classes.tableCell}>{errorHome}</TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </Grid>
    );
  }
}

MlbSideTable.propTypes = {
  data: PropTypes.shape({
    away_batter_totals: PropTypes.shape({
      runs: PropTypes.number.isRequired,
      hits: PropTypes.number.isRequired,
    }).isRequired,
    home_batter_totals: PropTypes.shape({
      runs: PropTypes.number.isRequired,
      hits: PropTypes.number.isRequired,
    }).isRequired,
    away_errors: PropTypes.number.isRequired,
    home_errors: PropTypes.number.isRequired,
  }).isRequired,
};

export default withStyles(styles)(MlbSideTable)