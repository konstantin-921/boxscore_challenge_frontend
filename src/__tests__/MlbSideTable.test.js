import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import MlbSideTable from '../components/MlbSideTable'

describe('MlbSideTable component', () => {

  const props = {
    away_batter_totals: { runs: 4, hits: 10 },
    home_batter_totals: { runs: 5, hits: 12 },
    away_errors: 3,
    home_errors: 2,
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(<MlbSideTable data={props} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  });

})