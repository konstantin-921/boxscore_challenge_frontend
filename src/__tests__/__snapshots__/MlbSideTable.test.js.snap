// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MlbSideTable component renders correctly 1`] = `
<div
  className="MuiGrid-item-3 MuiGrid-grid-xs-3-33"
>
  <table
    className="MuiTable-root-99"
  >
    <thead
      className="MuiTableHead-root-100"
    >
      <tr
        className="MuiTableRow-root-101 MuiTableRow-head-104"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-106 MuiTableCell-head-107 MlbSideTable-tableCell-1"
          scope="col"
        >
          R
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-106 MuiTableCell-head-107 MlbSideTable-tableCell-1"
          scope="col"
        >
          H
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-106 MuiTableCell-head-107 MlbSideTable-tableCell-1"
          scope="col"
        >
          E
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root-118"
    >
      <tr
        className="MuiTableRow-root-101"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-106 MuiTableCell-body-108 MlbSideTable-tableCell-1"
        >
          4
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-106 MuiTableCell-body-108 MlbSideTable-tableCell-1"
        >
          10
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-106 MuiTableCell-body-108 MlbSideTable-tableCell-1"
        >
          3
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-101"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-106 MuiTableCell-body-108 MlbSideTable-tableCell-1"
        >
          5
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-106 MuiTableCell-body-108 MlbSideTable-tableCell-1"
        >
          12
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-106 MuiTableCell-body-108 MlbSideTable-tableCell-1"
        >
          2
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
